### PR TITLE
Yuno break fix

### PR DESCRIPTION
**Pull request by anmol26s**
## Problem
- *Package remove python3-gevent-websocket and python3-gevent thus breaking YunoHost*

## Solution
- *Add packages in dependencies*

## PR Status
- [ ] Code finished.
- [ ] Tested with Package_check.
- [ ] Fix or enhancement tested.
- [ ] Upgrade from last version tested.
- [ ] Can be reviewed and tested.

## Package_check results
---
* An automatic package_check will be launch at https://ci-apps-dev.yunohost.org/, when you add a specific comment to your Pull Request: "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!"*

### CODE REVIEW

**Comment by anmol26s** [comment]
!testme

**Comment by anmol26s** [comment]
!testme

File: scripts/install [hunk 1]
```diff
@@ -110,10 +103,9 @@ ynh_add_nginx_config
 #=================================================
 ynh_script_progression --message="Building zeronet..."
 
-pushd "$final_path"
-	pip3 install --upgrade pip
-	pip3 install -r requirements.txt
-popd
+ynh_install_app_dependencies $pkg_dependencies
```
Comment (yalh76) [inline scripts/install:106]: Would be better to keep that in the `INSTALL DEPENDENCIES` step

File: scripts/install [hunk 2]
```diff
@@ -110,10 +103,9 @@ ynh_add_nginx_config
...
+pip3 install --upgrade pip
+pip3 install merkletools PySocks websocket_client gevent-ws coincurve bencode
```
Comment (yalh76) [inline scripts/install:108]: Why not directly using the requirements from `requirements.txt`

File: scripts/remove [hunk 3]
```diff
@@ -49,6 +49,7 @@ ynh_remove_systemd_config
 ynh_script_progression --message="Removing dependencies..."
 
 # Remove metapackage and its dependencies
+pip3 uninstall -y  merkletools PySocks coincurve maxminddb bencode
```
Comment (yalh76) [inline scripts/remove:52]: Not sure it's a good idea to remove them if they are use by other apps or YunoHost

File: scripts/restore [hunk 4]
```diff
@@ -85,11 +85,9 @@ ynh_install_app_dependencies $pkg_dependencies
 # BUILDING ZERONET
 #=================================================
 ynh_script_progression --message="Building zeronet..."
+pip3 install --upgrade pip
+pip3 install  merkletools PySocks websocket_client gevent-ws coincurve bencode
```
Comment (yalh76) [inline scripts/restore:89]: Why not directly using the requirements from `requirements.txt`

File: scripts/upgrade [hunk 5]
```diff
@@ -118,10 +118,8 @@ ynh_install_app_dependencies $pkg_dependencies
 #=================================================
 ynh_script_progression --message="Building zeronet..."
 
-pushd "$final_path"
-	pip3 install --upgrade pip
-	pip3 install -r requirements.txt
-popd
+pip3 install --upgrade pip
+pip3 install  merkletools PySocks websocket_client gevent-ws coincurve bencode
```
Comment (yalh76) [inline scripts/upgrade:122]: Why not directly using the requirements from `requirements.txt`

**Review by yalh76 — CHANGES_REQUESTED** [review]

**Comment by yalh76** [comment]
!testme

**Comment by anmol26s** [comment]
`Would be better to keep that in the INSTALL DEPENDENCIES step`
Its because that the requirement package do not mess with Yunohost dependencies. Installing from pip can break the Yunohost.
Some packages need to be secured through dependency helper so that Yunohost don't break.  
See this https://forum.yunohost.org/t/attributeerror-module-moulinette-interfaces-has-no-attribute-api/15724

And if additional dependencies come in we need to install it with pip anyways. 
https://forum.yunohost.org/t/zeronet-decentralized-websites/2584/8?u=kanhu


`Not sure it's a good idea to remove them if they are use by other apps or YunoHost`
Why keep the packages, when its not required by any app? Maybe by chance some apps uses it, app can be forced upgraded. 
Is there better way to check if the package is required by other app ?

**Comment by yalh76** [comment]
> 
> 
> `Would be better to keep that in the INSTALL DEPENDENCIES step`
> Its because that the requirement package do not mess with Yunohost dependencies. Installing from pip can break the Yunohost.
> Some packages need to be secured through dependency helper so that Yunohost don't break.
> See this https://forum.yunohost.org/t/attributeerror-module-moulinette-interfaces-has-no-attribute-api/15724

I agree with that but there is no difference between keeping it in it's own step `INSTALL DEPENDENCIES` like in example_ynh or having it in `BUILDING ZERONET`. For example during upgrade you didn't move the line `ynh_install_app_dependencies $pkg_dependencies`

> 
> And if additional dependencies come in we need to install it with pip anyways.
> https://forum.yunohost.org/t/zeronet-decentralized-websites/2584/8?u=kanhu
> 
> `Not sure it's a good idea to remove them if they are use by other apps or YunoHost`
> Why keep the packages, when its not required by any app? Maybe by chance some apps uses it, app can be forced upgraded.
> Is there better way to check if the package is required by other app ?

I don't know if there is a better way to check if python package are use somewhere but I think it's better to have unsued package than breaking other apps ...

**Comment by anmol26s** [comment]
Ok, so till we implement better way to handle python dependencies we can have any solution which don't break Yunohost.
Closing this.

**Pull request closed without merging**